### PR TITLE
Create servlet to determine login status of user, add content to frontend through JS.

### PR DESCRIPTION
**Pull request by zpChris**
Create the `AuthServlet.java` file (with much of the content taken from `HomeServlet.java`, a file in the walkthrough) that determines the user login status (and displays email, if applicable). The login / logout buttons and text are displayed to the frontend, through a manner similar to the comment system.

### CODE REVIEW

**Comment by zpChris** [comment]
Much of this work is actually proving outdated as a result of my new CL; I will merge that with this to remove the need to review old changes.

**Comment by zpChris** [comment]
I have merged CL #19 into this branch and CL. Normally, I would've requested reviewers on that CL, but it is actually removing / restructuring most of the work on this branch, so I figured the reviewers could just be on this branch (you can see the original CL #19 for more info).

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -186,6 +191,41 @@ function checkSubmit(inputElementId, submitElementId) {
 }
```
Comment (tristan-jh) [inline portfolio/src/main/webapp/script.js:191]: the walkthrough mentions preventing users who haven't logged in from leaving a comment. Definitely not required that you do that in this CL, but is that something you're planning to do in the future?

Comment (tristan-jh) [reply portfolio/src/main/webapp/script.js:191]: Disregard. Just saw pr #20

**Comment by zpChris** [comment]
Thanks for the question @tristan-jh -- I hadn't thought about whether the login system would still work when deployed, but I just deployed it for temporary testing and it works! (I got those instructions from the walkthrough, I didn't create that process of log in and the corresponding URL.)

File: portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      this(false, loginUrl, "", "");
+    }
+
+    // Constructor to create UserAuth object with user logged in.
+    public UserAuth(String logoutUrl, String email) {
+      this(true, "", logoutUrl, email);
+    }
+
+    // Full constructor to assign values to all fields.
+    public UserAuth(boolean loggedIn, String loginUrl, String logoutUrl,
```
[the remark this answers is not in the document]
Comment (zpChris) [reply portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java]: Thanks for the note -- I've updated multiple other methods to be `private` as well, as I realized they didn't require a `public` access level.

File: portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    UserService userService = UserServiceFactory.getUserService();
+    UserAuth userAuth;
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (bheff-work) [inline portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java:37]: nit: I think these would be better as constants

Comment (zpChris) [reply portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java]: Done!

**Pull request merged**